src/CodeGen_ARM.cpp

```diff
@@ -1162,6 +1162,12 @@ void CodeGen_ARM::codegen_vector_reduce(const VectorReduce *op, const Expr &init
             if (!i.defined()) {
                 i = make_zero(op->type);
             }
+            if (const Shuffle *s = matches[0].as<Shuffle>()) {
+                if (s->is_broadcast()) {
+                    // LLVM wants the broadcast as the second operand.
```

Comment:
Maybe augment this comment with text from the PR description to underscore that this is critical for dot-product generation.